tests/test_engine.py

```diff
@@ -52,6 +52,18 @@ def test_get_lookup_url(self):
         assert final_uri_with_params == '1/2/?extra_param=3'
         SampleResourceModel._lookup_urls = []
 
```

Comment:
I think we could use a few more tests similar to this where we explicitly add the same parameters in different orders but still get the same URL every time.